src/Illuminate/Foundation/Testing/TestCase.php

```diff
@@ -65,6 +65,8 @@ abstract class TestCase extends BaseTestCase
      */
     protected function setUp(): void
     {
+        Facade::clearResolvedInstances();
+
         if (! $this->app) {
             $this->refreshApplication();
```

Comment:
@taylorotwell Probably the `Facade::clearResolvedInstances();` should be inside this if condition. Not sure.

Reply:
I don't think so. The facade root acts in isolation as it's mostly static properties which keep instances in memory. I'm also not 100% sure about the side effects of this though.